pkg/operator/controller/ingress/metrics.go

```diff
@@ -15,6 +17,49 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
+var (
+	// ingressControllerConditions reports the status conditions of each
+	// IngressController using the ingress_controller_conditions metric.
+	ingressControllerConditions = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
This metric seems like it might have high cardinality. Do we really want to report true/false for every ingress controller status condition? For the purpose of alerting, we only need a small subset of status conditions to be reported via metrics, right?

At the very least, I think we can ignore the canary status condition here since that's reported as it's own metric.

Thoughts?